internal/declarative/v2/reconciler_test.go

```diff
@@ -156,3 +157,42 @@ func Test_hasDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestReconciler_hasStatusDiff_WhenSameStatus(t *testing.T) {
```

Comment:
I would add a test case for an empty (uninitialized) LastOperation. And maybe consider then joining the three test cases into a single table-driven test.